app-android/src/main/java/com/ustadmobile/port/android/view/WorkspaceEnterLinkFragment.kt

```diff
@@ -0,0 +1,111 @@
+package com.ustadmobile.port.android.view
+
+import android.os.Bundle
+import android.os.Handler
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.VisibleForTesting
+import androidx.core.util.PatternsCompat
+import com.toughra.ustadmobile.R
+import com.toughra.ustadmobile.databinding.FragmentWorkSpaceEnterLinkBinding
+import com.ustadmobile.core.controller.WorkspaceEnterLinkPresenter
+import com.ustadmobile.core.controller.WorkspaceEnterLinkPresenter.Companion.VALID_WORKSPACE_DOMAIN
+import com.ustadmobile.core.impl.UMAndroidUtil
+import com.ustadmobile.core.util.ext.toStringMap
+import com.ustadmobile.core.view.WorkspaceEnterLinkView
+import org.jetbrains.annotations.TestOnly
+
+
+class WorkspaceEnterLinkFragment : UstadBaseFragment(), WorkspaceEnterLinkView{
+
+    private var mBinding: FragmentWorkSpaceEnterLinkBinding? = null
+
+    private var mPresenter: WorkspaceEnterLinkPresenter? = null
+
+    private val inputCheckDelay: Long = 250
+
+    private val inputCheckHandler: Handler = Handler()
+
+    private var lastInputTime: Long = 0
+
+    @VisibleForTesting
+    private var isIdle = false
+
+    private val inputCheckerCallback = Runnable {
+        if (System.currentTimeMillis() > lastInputTime + inputCheckDelay) {
```

Comment:
Why do we need to check the time again here? Changing the text would have canceled the callback. If there was something wrong with the calculation, checkLinkValidity would never be called.